libraries/Wire/Wire_nRF52.cpp

```diff
@@ -394,16 +405,29 @@ void TwoWire::onService(void)
   }
 }
 
-TwoWire Wire(NRF_TWIM1, NRF_TWIS1, SPIM1_SPIS1_TWIM1_TWIS1_SPI1_TWI1_IRQn, PIN_WIRE_SDA, PIN_WIRE_SCL);
+TwoWire Wire(NRF_TWIM0, NRF_TWIS0, SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0_IRQn, PIN_WIRE_SDA, PIN_WIRE_SCL);
```

Comment:
This will be a breaking change, should I suggest that remain `NRF_TWIM1` etc here, so that `Wire` and `SPI` libs can be used at the same time. For `Wire1` using `NRF_TWIM0` etc. is fine.

Reply:
The micro:bit v2, unlike the original micro:bit, uses two separate i2c busses: one for the internal i2c devices and one for external devices, so you need both since the user might want to use the internal accelerometer and an external i2c device at the same time. However, I took a quick look at the nRF52833 manual and it appears to have three dedicated SPI peripherals and the nRF51 has two. So it may be possible to use the SPI peripheral for the SPI library on nRF52 boards. I'm not sure how bit a job that would be to implement.

Reply:
Ok, I think you just need to revert the changes you made to `Wire` to maintain backwards compatibility and have `Wire1` as:

```
TwoWire Wire1(NRF_TWIM0, NRF_TWIS0, SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0_IRQn, PIN_WIRE_SDA, PIN_WIRE_SCL);
```

Reply:
My understanding of the low-level details is limited, especially the interaction between SPI and TWI... 

Are you saying to use NRF_TWIM0, NRF_TWIS0, and SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0_IRQn for Wire1 and  NRF_TWIM1, NRF_TWIS1, and SPIM1_SPIS1_TWIM1_TWIS1_SPI1_TWI1_IRQn for Wire0? Is the issue that the interrupt vector is shared between TWI and SPI?

It might be best for you to make this change. The solution should allow use of both Wire and Wire1 at the same time (or, more accurately, interleaved). Ideally, it would also allow use of SPI as well.